core/src/logging/LogMasker.ts

```diff
@@ -18,99 +18,86 @@
-
-  protected readonly regex: RegExp = new RegExp(LogMaskingRegex.AUTH_USERNAME_REGEX)
-
-  static get instance (): AuthUsernameMask {
-    return this._instance
+function maskRequestConfigHeaders (headers: AxiosRequestHeaders): AxiosRequestHeaders {
+  const clonedHeaders: AxiosRequestHeaders = new AxiosHeaders()
+  for (const header in headers) {
+    clonedHeaders[header] = LOG_MASKING_HEADERS.includes(header.toLowerCase())
+      ? LoggingMessage.OMITTED
+      : headers[header]
   }
+  return clonedHeaders
 }
 
-class AuthPasswordMask extends Mask {
-  private static readonly _instance: AuthPasswordMask = new AuthPasswordMask()
-
-  protected readonly regex: RegExp = new RegExp(LogMaskingRegex.AUTH_PASSWORD_REGEX)
-
-  static get instance (): AuthPasswordMask {
-    return this._instance
+function maskAuthCredentials (auth: AxiosBasicCredentials | undefined): AxiosBasicCredentials | undefined {
+  if (auth === undefined) {
+    return auth
   }
-}
-
-class AccessTokenMask extends Mask {
-  private static readonly _instance: AccessTokenMask = new AccessTokenMask()
-
-  protected readonly regex: RegExp = new RegExp(LogMaskingRegex.ACCESS_TOKEN_REGEX)
-
-  static get instance (): AccessTokenMask {
-    return this._instance
+  return {
+    username: LoggingMessage.OMITTED,
+    password: LoggingMessage.OMITTED
   }
 }
 
-class NumberFieldMask extends Mask {
-  private static readonly _instance: NumberFieldMask = new NumberFieldMask()
-
-  protected readonly regex: RegExp = new RegExp(LogMaskingRegex.NUMBER_FIELD_REGEX)
-
-  static get instance (): NumberFieldMask {
-    return this._instance
+function maskResponseHeaders (headers: ResponseHeaders): ResponseHeaders {
+  const clonedHeaders: ResponseHeaders = {}
+  for (const header in headers) {
+    clonedHeaders[header] = LOG_MASKING_HEADERS.includes(header.toLowerCase())
+      ? LoggingMessage.OMITTED
+      : headers[header]
   }
+  return clonedHeaders
 }
 
-class PCIFieldsMask extends Mask {
-  private static readonly _instance: PCIFieldsMask = new PCIFieldsMask()
-
-  protected readonly regex: RegExp = new RegExp(LogMaskingRegex.PCI_FIELDS_REGEX)
-
-  static get instance (): PCIFieldsMask {
-    return this._instance
+function maskBodyFields (body: any): any {
+  if (typeof body !== 'object') {
+    return body
   }
+  for (const field in body) {
+    body[field] = (LOG_MASKING_BODY_FIELDS.includes(field.toLowerCase()) || isNumberField(field, body[field]))
+      ? LoggingMessage.OMITTED
+      : maskBodyFields(body[field])
+  }
+  return body
 }
 
-// eslint-disable-next-line @typescript-eslint/no-extraneous-class
-class MaskProvider {
-  private static readonly _masks: Mask[] = [AuthTokenMask.instance, AuthUsernameMask.instance, AuthPasswordMask.instance, AccessTokenMask.instance, NumberFieldMask.instance, PCIFieldsMask.instance]
-
-  static get masks (): Mask[] {
-    return this._masks
+function isNumberField (field: string, value: any): boolean {
```

Comment:
Let's rename it to `isPaymentCardNumber` or something similar.

Reply:
Done ✅